### ISSUE TITLE
Other backup places

### ISSUE THREAD

**Issue by Mike710Shine** [twofas/2fas-android#74]
It would be nice if an backup can be done by webdav of an nextcloud for example. I don't wan't do an update to google drive even is an hiden folder. Or an possability to deactivate the reminder of the backup.

**Comment by UlyssesZh** [comment]
Please provide other means of backing up tokens. Google Drive is proprietary software owned and hosted by Google. We need to be able to back our private information to self-hosted alternatives like Nextcloud.

**Comment by Schluggi** [comment]
Yes. Webdav & local would be nice.

**Comment by Shahin-rmz** [comment]
Plus one for this feature.
As a person who tries to degoogle his life, saving my password hashes inside Google Drive is no go.

**Comment by kdekooter** [comment]
+100 from another degoogler!

**Comment by CodeBleu** [comment]
Forgot to search before creating my own issue, but here is the link to it with my thoughts.

https://github.com/twofas/2fas-android/issues/112

`$ rm -rf bigtech`

**Comment by vedmant** [comment]
Would be nice to have a selection where to sync. For example options like Google Drive (on iOS), Dropbox, Nextcloud, etc... The file ofcource should be encrypted by a password. 

The only issue I can see is that user may forget the encryption password, probably makes sense to have a popup once in a while with password prompt, if user fails to type correct password, offer to set a new one (will require to update it on other devices as well).

**Comment by Th0masL** [comment]
I think adding other backup options than Google is indeed a must, specially for an product that claim that is security oriented, and respects user's privacy.

I'm trying to find a replacement from Authy, as they are deprecating their Desktop App by the end of the month.

It seems like many people have been recommending switching to 2FAS, but since I don't use a Google account on my GrapheneOS-based Android phone, I can't backup my keys, so that's a no go for me.

**Comment by CodeBleu** [comment]
> .... I can't backup my keys, so that's a no go for me.
> 

FYI - Although it is correct there is no other option to do a "sync" backup to somewhere other than Google, you can still manually export everything  and put it wherewver you like.

**Comment by Lindsor** [comment]
I agree on alternate sync locations.
Especially since that would open the door to shared sync between android/ios device.
I have 2 devices that i use for work/personal and i need my 2fas on both, currently there is no way of doing this.

Im also looking at moving away from Authy and this is a blocker for me

**Comment by elliotwutingfeng** [comment]
2FAS should at least allow automated versioned backups to the local phone filesystem, like Aegis.

**Comment by LanceLake** [comment]
+1 to this. Let me save it to my local NAS so I can control where it goes. There is no promise from google that drive won't be taken away or your account being shut down.

**Comment by Tmalll** [comment]
Chinese users using Chinese mobile phones say that Google backup is very unfriendly because all Google services are blocked in China, damn GFW. There is not even gms service on mobile phones in China. I agree with webdav's solution, or you can add onedrive, which can still be used in mainland China. Or another cloud storage provider, as long as the setup isn't that complicated, since I'm a newbie.

**Comment by SynapticWanderer** [comment]
as another degoogler I also want this very much, would accept Dropbox if necessary, but would prefer Proton Drive

**Comment by storm49152** [comment]
While implementing any of the big-tech-specific backup backends would be nice, since it helps a lot of people who cannot or don't want to use Google, personally, I would vote to include WebDAV, as it is the most generic solution that would work with most WebDAV server implementations.

**Comment by tomme87** [comment]
This is now the only reason I have not fully committed to 2FAS yet. I would prefer Proton Drive, but even automated backups to the local file system would be better. I don't want to use Google drive.

**Comment by SynapticWanderer** [comment]
yeah me too, if I wanted to use google products for my authenticator I would still be using their authenticator instead of this one

**Comment by Tmalll** [comment]
+10086

**Comment by avesst** [comment]
+100

I have my own server/NAS at home and would like to sync to that using something like SMB/WebDAV/rsync/SFTP.

Or as someone else requested, let me backup locally to my phone and I can sync it using whatever software I'd like.

Google only alternative rhymes uneasy with the open source philosophy imo.

**Comment by Q303835** [comment]
I strongly support this request. Adding WebDAV and S3-compatible storage (e.g., Cloudflare R2, AWS S3, MinIO) would be a huge improvement for 2FAS.

Key reasons:

S3 Support: Many users now prefer S3-compatible buckets for their reliability and cost-effectiveness (like Cloudflare R2).

De-Googling: Essential for users who don't use GMS or prefer self-hosted solutions like Nextcloud.

Privacy: Gives users full control over their encrypted vault location.

This feature would make 2FAS the most versatile 2FA app on Android. Looking forward to it!

**Comment by 43976989** [comment]
Adding to this: there should be a way to disable the prompt. Currently the only option is to use the platform native, and it nags continuously about it. Yes, it's a risk to not backup but users should be able to make that choice.

**Comment by justindemuro** [comment]
Anyone want to band together and just "do it ourselves"? The source code is open. I would put effort into adding a few lines that allow at least automating a backup file into the local filesystem at minimum.

**Comment by LanceLake** [comment]
> Anyone want to band together and just "do it ourselves"? The source code is open. I would put effort into adding a few lines that allow at least automating a backup file into the local filesystem at minimum.

I would, but I don't know the language well enough to do it.

**Comment by 43976989** [comment]
> I would, but I don't know the language well enough to do it.

Likewise - Kotlin in particular isn't one I have any familiarity with, and especially given the nature of the application I don't want to submit junk.
